common-app-lib/test/cmr/common_app/test/api/launchpad_token_validation_tests.clj

```diff
@@ -0,0 +1,52 @@
+(ns cmr.common-app.test.api.launchpad-token-validation-tests
+  "Unit tests for health checks"
+  (:require
+   [clojure.data.codec.base64 :as base64]
+   [clojure.test :refer :all]
+   [cmr.common-app.api.launchpad-token-validation :as token]))
+
+(defn- test-matrix
+  "The input-list, message-list, and expected-list are all seq of the same size.
+   Items match in each seq and corrispond to the same test. 'input' is passed
+   to 'tester' and then tested with '=' against 'expected'. If the test fails
+   then 'message' is sent back to the test API. Parameter order allows for the
+   use of partial to fix parameters expected to change the least from one test
+   to the next"
+  [input-list message-list expected-list tester]
+  (doseq [[expected input message] (map list expected-list input-list message-list)]
+    (is (= expected (tester input)) (str "Failed: " message " not " expected))))
+
+
+(defn random-base64
+  [len]
+  (let [junk (apply str (take len (repeatedly #(char (+ (rand 26) 65)))))]
+    (String. (base64/encode (.getBytes junk)) "UTF-8")))
+
+(deftest token-tests
+  ;; Security Note, these are valid looking tokens but are not and never were
+  ;; active tokens. Can be generated with something like SecureRandom.hex 32
+  ;; under rails
+  (let [token-legacy "78221B66-EF2B-9283-19B5-295D446B065B"
+        token-oauth "EDL-O84a9c44c0ad2c628300477aa2568ba39e2fa47190f5dc16f95fb24086f7"
+        token-user "EDL-U84a9c44c0ad2c628300477aa2568ba39e2fa47190f5dc16f95fb24086f7"
+        token-client "EDL-C84a9c44c0ad2c628300477aa2568ba39e2fa47190f5dc16f95fb24086f7"
+        token-ropc "EDL-R84a9c44c0ad2c628300477aa2568ba39e2fa47190f5dc16f95fb24086f7"
+        token-bearer (str "Bearer " token-user)
```

Comment:
Is it possible for us to have a Bearer token not starts with `Bearer EDL-`? I think that is possible. If so, I would like to see it in the test.

Reply:
I can add in the case where client id is passed in, I think that is what you mean, the EDL-U test is already there

Reply:
added new tests and I formatted the tests so that it, IMHO, it is easy to read.